File: problemSolving/java/programmers/크레인_인형뽑기_게임.java
import java.util.ArrayList;

class Solution {
    public int solution(int[][] board, int[] moves) {
        ArrayList<Integer> list = new ArrayList<Integer>();
        int answer = 0;
        for (int i = 0; i < moves.length; i++) {
            int now = moves[i] - 1;
            int num = 0;
            for (int j = 0; j < board.length; j++) {
                if (board[j][now] > 0) {
                    num = board[j][now];
                    board[j][now] = 0;
                    break;
                }
            }
            if (num > 0) {
                if (list.size() > 0) {
                    if (list.get(list.size() - 1) == num) {
                        answer += 2;
                        list.remove(list.size() - 1);
                    } else {
                        list.add(num);
                    }
                } else {
                    list.add(num);
                }
            }
        }
        return answer;
    }
}

File: problemSolving/programmers/level_1_js/크레인인형뽑기게임.js
function solution(board, moves) {
  var answer = 0;
  var basket = new Array();
  for (var i in moves) {
    for (var j = 0; j < board.length; j++) {
      if (board[j][moves[i] - 1]) {
        if (basket.length > 0) {
          if (basket[basket.length - 1] === board[j][moves[i] - 1]) {
            answer += 2;
            board[j][moves[i] - 1] = 0;
            basket.pop();
            break;
          }
        }
        basket.push(board[j][moves[i] - 1]);
        board[j][moves[i] - 1] = 0;
        break;
      }
    }
  }
  return answer;
}
